lib/liquid/block_body.rb

```diff
@@ -1,6 +1,7 @@
 module Liquid
   class BlockBody
-    FullToken = /\A#{TagStart}#{WhitespaceControl}?\s*(\w+)\s*(.*?)#{WhitespaceControl}?#{TagEnd}\z/om
+    LiquidTagToken = /\A\s*(\w+)\s*(.*?)\z/o
+    FullToken = /\A#{TagStart}#{WhitespaceControl}?(\s*)(\w+)(\s*)(.*?)#{WhitespaceControl}?#{TagEnd}\z/om
```

Comment:
This is change is definitely debatable. Without it, we don't capture whitespace between `{%` and the tag name, and thus can't calculate line numbers within the tag. We really want good errors with line numbers inside the `{% liquid` tag though. Is it worth the code complexity that'll come from making a second pattern without the whitespace capture groups, to use when line numbers are off?

Reply:
> Since this constant is publicly visible, there are chances that this constant is used in third-party Liquid plugins in the form of **`raw.scan(Liquid::BlockBody::FullToken)`** or `text =~ Liquid::BlockBody::FullToken; markup = $2`

This is the problem with ruby making things public by default and inconvenient to make private.  I don't think we actually intended for this to be used externally.  In fact, we don't have a single `private_constant` in the codebase, yet we treat these as internal.

I think we are more concerned about compatibility with existing liquid code which isn't a concern here.  Perhaps we should just mark this regex as `private_constant` so breakage is clearer to any third-party code that happens to be using it.